src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/Profile.vue

```diff
@@ -162,17 +169,21 @@ import type { Components } from "@/types/openapi";
 import { formatDate } from "@/utils/format";
 import { formatPhoneNumber } from "@/utils/format";
 import { areObjectsEqual } from "@/utils/functions";
+import { useLoadingStore } from "@/store/loading";
+import Loading from "@/components/Loading.vue";
 
 export default defineComponent({
   name: "Profile",
-  components: { PageContainer, LinkBar, Callout },
+  components: { PageContainer, LinkBar, Callout, Loading },
   setup: async () => {
-    const userProfile = await getProfile();
     const userStore = useUserStore();
-
-    userStore.setUserProfile(userProfile);
-
-    return { userProfile, userStore };
+    const loadingStore = useLoadingStore();
+    let userProfile = userStore.userProfile;
+    onMounted(async () => {
+      userProfile = await getProfile();
+      userStore.setUserProfile(userProfile);
+    });
+    return { userProfile, loadingStore, userStore };
```

Comment:
Pattern question, 

In the past we've put the mounted() outside the setup section to follow the options api pattern. 

Example: 
![image](https://github.com/user-attachments/assets/510cb71e-0228-4b06-ae07-8acc7eb5296b)

https://github.com/bcgov/ECC-ECER/blob/ed6f6f8669931f563a9de37b3957f4d3584058ab/src/ECER.Clients.RegistryPortal/ecer.clients.registryportal.client/src/components/CertificationCard.vue#L90

Would suggest following this same pattern so code is consistent. I will say both ways work.